_posts/2020-06-18-a-sustainable-nuget-marketplace-will-have-to-compete-with-the-nuget-gallery.markdown

```diff
@@ -0,0 +1,368 @@
+
+The legal aspect will be a bit like what the open source foundations offer:
+help with protecting producers and consumers.
+
+There is one additional aspect a NuGet marketplace will have to help with: establishing rules and preventing
+their abuse. What if I have an MIT-licensed project, and someone sells it with a dual license on this
+marketplace?
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">It will be perfectly legal and valid for someone to sell your MIT licence code. You expressly permitted that.</p>&mdash; Damian Hickey (@randompunter) <a href="https://twitter.com/randompunter/status/1273173878752370688?ref_src=twsrc%5Etfw">June 17, 2020</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+Morally, this is "not done", but legally, this may happen. The marketplace will have to prevent this and have
+rules and guidance on how to go about this. We want sustainability!
+
+Another similarity with open source foundations is in the legality of purchasing packages.
+A NuGet marketplace will have to make sure consumers who purchase here can trust their purchase is legal
+and they can use the package without issue, now and in the future.
+
+Others are GDPR (as a consumer, is my personal data safe?), and probably some more. Feel free to add them
+in the comments and I will update the post.
+
+### For producers: business services
+
+Before we can sell packages, what's the legal status of an open source project? Is it a company? A non-profit?
+
+A NuGet marketplace will probably be the legal entity where consumers purchase packages from.
+Much like how [Fastspring](https://www.fastspring.com/) acts as a reseller, a NuGet marketplace will have to
+do this as well.
+
+A NuGet marketplace will have to act as a reseller, for several reasons:
+
+* Consumers purchase from one legal entity, with one contract.
+* Consumers can have e.g. a subscription that includes paid-for open source from multiple vendors, without dealing with all of them separately.
+* Consumers will be in several locations - the marketplace will have to accomodate for Value Added Tax (VAT), the EU VAT MOSS, and all that.
+
+If we want to make the step to sustainable open source, a NuGet marketplace will have to help
+projects in doing so. Not simply by having a downloadable binary with a paywall, but by helping them out
+with being a business.
+
+### For producers and consumers: flexible licensing and purchasing options
+
+A NuGet marketplace will need to support various purchasing options. Some examples:
+
+* Trial versions (30 days of usage)
+* Subscriptions (containing one or multiple packages)
+* Licenses with # seats, ideally also where enterprises can manage who can consume packages
+* Perpetual licenses
+* Licenses that are valid for major versions of a package only, etc.
+
+And as a consumer, can we pay by credit card only? Raise a purchase order and have NET90 payment terms?
```

Comment:
That'd depend on the size of the purchase. No one is going to bother doing NET90 with POs for a $99 purchase.